### PR TITLE
travis: reduce the number of travis builders

### PR DESCRIPTION
**Pull request by tamird**
This collapses all the "platform dependent" tests into a single travis
builder in an effort to reduce overall CI times. These builds currently
take a combined 21-23 minutes, but each one has to compile the library,
so combining them should yield some time savings (5-10 minutes).

Unfortunately the other builders don't duplicate work, so combining
them is unlikely to provide benefit.

### CODE REVIEW

**Comment by yiwu-arbug** [comment]
@tamird let's wait for the travis build and see how it goes.

**Comment by tamird** [comment]
@yiwu-arbug of course.

By the way, it looks like you guys are currently aggressively maintaining two branches: master, and 5.4.fb. This is really unfortunate, because it adds to the already extremely congested Travis CI situation in this repo.

Could you turn off Travis CI on the non-master branches? It currently takes over a day to get a Travis build.

See https://travis-ci.org/facebook/rocksdb/builds (at the time of writing):

screen shot 2017-05-16 at 15 22 02

**Comment by yiwu-arbug** [comment]
@tamird 5.4.fb etc. are release branches. We need travis build there. We can probably turn off travis for other branches, but they are also useful in case people want to run ac-hoc travis job.

**Comment by tamird** [comment]
I understand that they're useful in general, but something needs to be done - the build takes over 4 hours on average and the congestion causes huge delays.

Perhaps you could do the cherry-picks to the release branches once a day, instead of eagerly whenever anything merges to master?

**Comment by yiwu-arbug** [comment]
We only cherry-pick what's needed into release branches. The thing is travis don't have enough resources and we don't have a better alternative to it..

**Comment by tamird** [comment]
Right, I understand that, but if you batched your cherry-picks at the end of the day, you would substantially reduce the congestion during daylight hours when people are trying to get work done.

This would prevent merge-on-red situations like the one caused by #2269, which has required at least 3 follow-up PRs now.

**Comment by tamird** [comment]
Heads up that this is green, and the build took 1h49m compared to the usual 3h29m:

screen shot 2017-05-25 at 11 06 26

There's a merge conflict with #2127 that I'm fixing now, but this should be good to go.

**Comment by tamird** [comment]
@yiwu-arbug can you take a look, please?

File: .travis.yml [hunk 1]
```diff
@@ -4,9 +4,6 @@ language: cpp
 os:
   - linux
   - osx
-compiler:
-  - clang
```
Comment (yiwu-arbug) [inline .travis.yml]: We want to keep clang build. How does it affect total time per your experiment?

**Comment by yiwu-arbug** [comment]
LGTM. @siying any comment?

**Comment by siying** [comment]
No I don't have any comment.
________________________________

File: .travis.yml (hunk 1 above)
Comment (yiwu-arbug) [inline .travis.yml]: platform_dependent is not building all tests, so there can be compile error catch by clang but not gcc. Hope this case can be covered.

Comment (tamird) [inline .travis.yml]: Ah, you mean that `TEST_GROUP=1` and `TEST_GROUP=2` don't run on OSX? I can enable those.

Comment (yiwu-arbug) [inline .travis.yml]: Or simply building the tests without running them.

Comment (tamird) [inline .travis.yml]: Ah, is there a make target that does that?

Comment (yiwu-arbug) [inline .travis.yml]: Not sure. You can do `make all` though.

Comment (yiwu-arbug) [inline .travis.yml]: @tamird seems TEST_GROUP 1&2 is taking an hour to run. Mind remove them and add `make all` for osx?

Comment (tamird) [inline .travis.yml]: Done!

**Comment by tamird** [comment]
Green again!

**Comment by yiwu-arbug** [comment]
Merging. Thank you for your contribution!

**Comment by tamird** [comment]
🎉

**Pull request closed without merging**